src/cascadia/UnitTests_TerminalCore/SelectionTest.cpp

```diff
@@ -672,5 +672,121 @@ namespace TerminalCoreUnitTests
             selection = term.GetViewport().ConvertToOrigin(selectionRects.at(1)).ToInclusive();
             VERIFY_ARE_EQUAL(selection, SMALL_RECT({ 0, 11, 99, 11 }));
         }
+
+        TEST_METHOD(ShiftClick)
+        {
+            Terminal term;
+            DummyRenderTarget emptyRT;
+            term.Create({ 100, 100 }, 0, emptyRT);
+
+            // set word delimiters for terminal
+            auto settings = winrt::make<MockTermSettings>(0, 100, 100);
+            term.UpdateSettings(settings);
+
+            // Insert text at position (4,10)
+            const std::wstring_view text = L"doubleClickMe dragThroughHere";
+            term.SetCursorPosition(4, 10);
+            term.Write(text);
+
+            // Step 1: Create a selection on "doubleClickMe"
+            {
+                // Simulate double click at (x,y) = (5,10)
+                term.MultiClickSelection({ 5, 10 }, Terminal::SelectionExpansionMode::Word);
+
+                // Validate selection area: "doubleClickMe" selected
+                ValidateSingleRowSelection(term, SMALL_RECT({ 4, 10, 16, 10 }));
+            }
+
+            // Step 2: Shift+Click to "dragThroughHere"
+            {
+                // Simulate Shift+Click at (x,y) = (21,10)
+                //
+                // buffer: doubleClickMe dragThroughHere
+                //         ^                ^
+                //       start            finish
+                term.SetSelectionEnd({ 21, 10 }, ::Terminal::SelectionExpansionMode::Cell);
+
+                // Validate selection area: "doubleClickMe drag" selected
+                ValidateSingleRowSelection(term, SMALL_RECT({ 4, 10, 21, 10 }));
+            }
+
+            // Step 3: Shift+Double-Click to "dragThroughHere"
+            {
+                // Simulate Shift+DoubleClick at (x,y) = (21,10)
+                //
+                // buffer: doubleClickMe dragThroughHere
+                //         ^                ^
+                //       start            finish
+                term.SetSelectionEnd({ 21, 10 }, ::Terminal::SelectionExpansionMode::Word);
+
+                // Validate selection area: "doubleClickMe dragThroughHere" selected
+                ValidateSingleRowSelection(term, SMALL_RECT({ 4, 10, 32, 10 }));
+            }
+
+            // Step 4: Shift+Double-Click to "dragThroughHere"
+            {
+                // Simulate Shift+TripleClick at (x,y) = (21,10)
+                //
+                // buffer: doubleClickMe dragThroughHere
+                //         ^                ^
+                //       start            finish
+                term.SetSelectionEnd({ 21, 10 }, ::Terminal::SelectionExpansionMode::Line);
+
+                // Validate selection area: "doubleClickMe dragThroughHere..." selected
+                ValidateSingleRowSelection(term, SMALL_RECT({ 4, 10, 99, 10 }));
+            }

```

Comment:
I'd like to see a test that does a simulates a shift double-click, drag, _drag back_.

I'd expect it to go from doubleClickMe to dragThroughHere, and then totally deselect dragThroughHere when you drag back.

Reply:
Good callout. It revealed a regression in the TermControl so I moved the touchdown pos back. 😊

Reply:
huh what was the issue?

Reply:
Shift+Click, then drag was broken.
Before latest commit: drag --> clear and start a new selection
now: drag --> only set the "end" of selection

I think I used to only test Shift+Click, then Shift+Click somewhere else.